Ktisis/Interop/PoseHooks.cs

```diff
@@ -0,0 +1,122 @@
+		private static Hook<SetBoneModelSpaceFfxivDelegate> SetBoneModelSpaceFfxivHook = null!;
+
+		private delegate IntPtr CalculateBoneModelSpaceDelegate(ref HkaPose pose, int boneIdx);
+		private static Hook<CalculateBoneModelSpaceDelegate> CalculateBoneModelSpaceHook = null!;
+		
+		private unsafe delegate void SyncModelSpaceDelegate(HkaPose* pose);
+		private static Hook<SyncModelSpaceDelegate> SyncModelSpaceHook = null!;
+		
+		private unsafe delegate hkQsTransform* AccessBoneModelSpaceDelegate(HkaPose* pose, int boneIdx, int propagate);
+		private static AccessBoneModelSpaceDelegate AccessBoneModelSpaceFunc = null!;
+		
+		private unsafe delegate hkQsTransform* AccessBoneLocalSpaceDelegate(HkaPose* pose, int boneIdx);
+		private static AccessBoneLocalSpaceDelegate AccessBoneLocalSpaceFunc = null!;
+		
+		internal static bool PosingEnabled { get; private set; }
+		
+		internal static unsafe void Init() {
+			var setBoneModelSpaceFfxiv = Dalamud.SigScanner.ScanText("48 8B C4 48 89 58 18 55 56 57 41 54 41 55 41 56 41 57 48 81 EC ?? ?? ?? ?? 0F 29 70 B8 0F 29 78 A8 44 0F 29 40 ?? 44 0F 29 48 ?? 48 8B 05 ?? ?? ?? ?? 48 33 C4 48 89 84 24 ?? ?? ?? ?? 48 8B B1");
+			SetBoneModelSpaceFfxivHook = Hook<SetBoneModelSpaceFfxivDelegate>.FromAddress(setBoneModelSpaceFfxiv, SetBoneModelSpaceFfxivDetour);
+			
+			var calculateBoneModelSpace = Dalamud.SigScanner.ScanText("40 53 48 83 EC 10 4C 8B 49 28");
+			CalculateBoneModelSpaceHook = Hook<CalculateBoneModelSpaceDelegate>.FromAddress(calculateBoneModelSpace, CalculateBoneModelSpaceDetour);
+
+			var syncModelSpace = Dalamud.SigScanner.ScanText("48 83 EC 18 80 79 38 00");
+			SyncModelSpaceHook = Hook<SyncModelSpaceDelegate>.FromAddress(syncModelSpace, SyncModelSpaceDetour);
+
+			// Not necessary for posing but useful for attempting bone parenting, check Havok docs
+			var accessBoneModelSpace = Dalamud.SigScanner.ScanText("48 8B C4 89 50 10 53 57");
+			AccessBoneModelSpaceFunc = Marshal.GetDelegateForFunctionPointer<AccessBoneModelSpaceDelegate>(accessBoneModelSpace);
+			
+			var accessBoneLocalSpace = Dalamud.SigScanner.ScanText("4C 8B DC 53 55 56 57 41 54 41 56 48 81 EC");
+			AccessBoneLocalSpaceFunc = Marshal.GetDelegateForFunctionPointer<AccessBoneLocalSpaceDelegate>(accessBoneLocalSpace);
+		}
+
+		internal static void DisablePosing()
+		{
+			CalculateBoneModelSpaceHook?.Disable();
+			SetBoneModelSpaceFfxivHook?.Disable();
+			SyncModelSpaceHook?.Disable();
+			PosingEnabled = false;
+		}
+
+		internal static void EnablePosing()
+		{
+			CalculateBoneModelSpaceHook?.Enable();
+			SetBoneModelSpaceFfxivHook?.Enable();
+			SyncModelSpaceHook?.Enable();
+			PosingEnabled = true;
+		}
+
+		/// <summary>
+		/// Toggles posing mode via hooks.
+		/// </summary>
+		/// <returns></returns>
+		internal static bool TogglePosing()
+		{
+			if (CalculateBoneModelSpaceHook.IsEnabled)
```

Comment:
Checking whether each hook is enabled individually looks like it might cause internal desyncs, especially if these methods can throw. (Though if they can, we should probably move the enable-bool changes further up)

Suggest just checking `PosingEnabled` instead of the individual hook status.
```diff
@@ -63,18 +63,15 @@ internal static void EnablePosing()
                /// <returns></returns>
                internal static bool TogglePosing()
                {
-                       if (CalculateBoneModelSpaceHook.IsEnabled)
+                       if(PosingEnabled) {
                                CalculateBoneModelSpaceHook.Disable();
-                       else
-                               CalculateBoneModelSpaceHook.Enable();
-                       if (SetBoneModelSpaceFfxivHook.IsEnabled)
                                SetBoneModelSpaceFfxivHook.Disable();
-                       else
-                               SetBoneModelSpaceFfxivHook.Enable();
-                       if (SyncModelSpaceHook.IsEnabled)
                                SyncModelSpaceHook.Disable();
-                       else
+                       } else {
+                               CalculateBoneModelSpaceHook.Enable();
+                               SetBoneModelSpaceFfxivHook.Enable();
                                SyncModelSpaceHook.Enable();
+                       }
                        PosingEnabled = !PosingEnabled;
                        return PosingEnabled;
                }
```